src/frontend/Pretty_printing.ml

```diff
@@ -379,33 +379,38 @@ and pp_block block_name ppf block_stmts =
 and pp_opt_block ppf block_name opt_block =
   Fmt.option ~none:Fmt.nop (pp_block block_name) ppf opt_block
 
-and pp_program ppf = function
-  | { functionblock= bf
+and pp_program ppf
+    { functionblock= bf
     ; datablock= bd
     ; transformeddatablock= btd
     ; parametersblock= bp
     ; transformedparametersblock= btp
     ; modelblock= bm
-    ; generatedquantitiesblock= bgq } ->
-      Format.pp_open_vbox ppf 0 ;
-      pp_opt_block ppf "functions" bf ;
-      pp_opt_block ppf "data" bd ;
-      pp_opt_block ppf "transformed data" btd ;
-      pp_opt_block ppf "parameters" bp ;
-      pp_opt_block ppf "transformed parameters" btp ;
-      pp_opt_block ppf "model" bm ;
-      pp_opt_block ppf "generated quantities" bgq ;
-      Format.pp_close_box ppf ()
+    ; generatedquantitiesblock= bgq } =
+  Format.pp_open_vbox ppf 0 ;
+  pp_opt_block ppf "functions" bf ;
+  pp_opt_block ppf "data" bd ;
+  pp_opt_block ppf "transformed data" btd ;
+  pp_opt_block ppf "parameters" bp ;
+  pp_opt_block ppf "transformed parameters" btp ;
+  pp_opt_block ppf "model" bm ;
+  pp_opt_block ppf "generated quantities" bgq ;
+  Format.pp_close_box ppf ()
 
 and pretty_print_program p =
   let result = wrap_fmt pp_program p in
-  let check_correctness () =
-    let result_ast =
-      Errors.without_warnings
-        (Parse.parse_string Parser.Incremental.program)
-        result
-    in
-    if compare_untyped_program p (Option.value_exn (Result.ok result_ast)) <> 0
-    then failwith "Pretty printing failed. Please file a bug."
-  in
-  check_correctness () ; result
+  (* let check_correctness () =
+   *   let result_ast =
+   *     Errors.without_warnings
+   *       (Parse.parse_string Parser.Incremental.program)
+   *       result
+   *   in
+   *   if
+   *     compare_untyped_program
+   *       (untyped_program_of_typed_program p)
+   *       (Option.value_exn (Result.ok result_ast))
+   *     <> 0
+   *   then failwith "Pretty printing failed. Please file a bug."
+   * in
+   * check_correctness () ; *)
```

Comment:
Why is this commented out?

Reply:
Oh, it wasn't working before because I started using the pretty printer on both typed and untyped programs and I think the correctness check was failing because the conversion to untyped was changing the tree. I think it will probably work now that your PR is in, can try it out.